### PR TITLE
AJ-1115 workload identity options

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/AJ-1115

Reminder:

PRs merged into main will not automatically generate a PR in https://github.com/broadinstitute/terra-helmfile to update the WDS image deployed to kubernetes - this action will need to be triggered manually by running the following github action: https://github.com/DataBiosphere/terra-workspace-data-service/actions/workflows/tag.yml. Dont forget to provide a Jira Id when triggering the manual action, if no Jira ID is provided the action will not fully succeed. 

After you manually trigger the github action (and it completes with no errors), you must go to [the terra-helmfile](https://github.com/broadinstitute/terra-helmfile) repo and verify that this generated a PR that merged successfully.

The terra-helmfile PR merge will then generate a PR in [leonardo](https://github.com/DataBiosphere/leonardo).  This will automerge if all tests pass, but if jenkins tests fail it will not; be sure to watch it to ensure it merges. To trigger jenkins retest simply comment on PR with "jenkins retest".

### CODE REVIEW

File: service/src/main/resources/application.properties [hunk 1]
```diff
@@ -4,7 +4,9 @@ management.endpoints.web.path-mapping.info=version
 management.endpoints.web.path-mapping.health=status
 management.endpoint.health.show-details=ALWAYS
 
-spring.datasource.jdbc-url=jdbc:postgresql://${WDS_DB_HOST:localhost}:${WDS_DB_PORT:5432}/wds?reWriteBatchedInserts=true
+# When running on Azure in k8s with workload identity set ADDITIONAL_JDBC_URL_PARAMS to
+# sslmode=require&authenticationPluginClassName=com.azure.identity.extensions.jdbc.postgresql.AzurePostgresqlAuthenticationPlugin
+spring.datasource.jdbc-url=jdbc:postgresql://${WDS_DB_HOST:localhost}:${WDS_DB_PORT:5432}/${DB_NAME:wds}?reWriteBatchedInserts=true${ADDITIONAL_JDBC_URL_PARAMS}
```
Comment (davidangb) [inline service/src/main/resources/application.properties:9]: we'll also need `{ADDITIONAL_JDBC_URL_PARAMS}` on the `streaming.query.jdbc-url` definition

**Review by davidangb — COMMENTED** [review]
one comment inline, this looks much simpler than I had expected/feared it might be!

I think we can get rid of the `ManagedIdentityDao` now too, if we don't need that code for this implementation.

**Comment by yuliadub** [comment]
what David said, my only other question is, what would provide the database endpoint url? would that be something Leo would give to us or is this an api WDS would have to make? since I am not seeing changes in any other db transactions besides cloning maybe that info is already in WDS somewhere and I just dont know where to look.

File: service/src/main/java/org/databiosphere/workspacedataservice/service/BackupService.java [hunk 2]
```diff
@@ -43,13 +46,16 @@ public class BackupService {
...
     public BackupResponse backupAzureWDS(BackUpFileStorage storage, String version) {
         try {
             validateVersion(version);
             String blobName = GenerateBackupFilename();
 
             List<String> commandList = GenerateCommandList();
-            Map<String, String> envVars = Map.of("PGPASSWORD", dbPassword);
+            Map<String, String> envVars = Map.of("PGPASSWORD", determinePassword());
 
             LocalProcessLauncher localProcessLauncher = new LocalProcessLauncher();
```
Comment (yuliadub) [inline service/src/main/java/org/databiosphere/workspacedataservice/service/BackupService.java:60]: today this list of commands also takes in the url to the db (which is localhost, set by env param) would the expectation be that we get the flex postgres endpoint from Leo?

Comment (dvoet) [reply service/src/main/java/org/databiosphere/workspacedataservice/service/BackupService.java:60]: yes, Leo would populate relevant env vars

**Review by davidangb — APPROVED** [review]
thank you!

**Review by yuliadub — APPROVED** [review]

**Comment by yuliadub** [comment]
thank you!  properties file looks so much better :)

**Pull request merged**